src/flow.h

```diff
@@ -475,6 +476,8 @@ typedef struct Flow_
     uint32_t tosrcpktcnt;
     uint64_t todstbytecnt;
     uint64_t tosrcbytecnt;
+
+    MacSet *macset;
```

Comment:
Since this behavior is optional, we're wasting 8 bytes per flow here if the feature is not used. Maybe this would be a good case for the `FlowStorage` API?

Reply:
I'll take a look.

Reply:
Hmm. So far only _logging_ of these values has been optional (via the `outputs.eve-log.ethernet` config item). Note that this can be disabled in one EVE output and enabled in another. Bookkeeping of the addresses is always performed up to now.

I have started to implement a solution that would check it at least one of the defined (enabled) EVE-JSON outputs has their `ethernet` setting set to true, and only then register a flow storage item for the `MacSet`s. All `MacSet*` functions would no longer directly be given a `MacSet*` pointer but a `Flow*`, and if there is no registered flow storage for `MacSet`s, all functions would immediately return without any action.
Any comments on this idea?

Reply:
Perhaps `ProvidesFeature` and `RequiresFeature` could be used here when `ehternet` logging is configured.

This is used currently in `output-filestore.c` and `detect-filestore.c` ... The former signals the feature is available and the second indicates the dependency if a rule is seen that requires it.